### PR TITLE
re merge my progress: monetization and transaction list

### PR DESCRIPTION
**Pull request by ricardocr987**

### CODE REVIEW

**Review by MHHukiewitz — APPROVED** [review]
Looks a lot simpler and cleaner than before. Much appreciated!

LGTM

File: src/pages/data/components/BrowseDataTable.tsx [hunk 1]
```diff
@@ -32,80 +19,35 @@ const BrowseDataTable = ({
...
-          );
-          setSignature(sig);
-        } catch (error) {
-          console.error('Error sending transaction:', error);
-        }
-      };
+    const serializedBuffer = Buffer.from(transaction, 'base64');
+    const unsignedTransaction = VersionedTransaction.deserialize(serializedBuffer);
+    const signedTransaction = await signTransaction(unsignedTransaction);
+    const signedSerializedBase64 = Buffer.from(signedTransaction.serialize()).toString('base64');
```
[the remark this answers is not in the document]
Comment (ricardocr987) [reply src/pages/data/components/BrowseDataTable.tsx:41]: The server sends serialized transactions as strings, it converts them to Transaction objects to be able to sing them, then just do the inverse operation to send the transaction just as a string. This is done to reduce the content of the message between server and client.

[the remark this answers is not in the document]
Comment (MHHukiewitz) [reply src/pages/data/components/BrowseDataTable.tsx:41]: Would it make sense to allow both Transaction objects and strings as inputs to `signTransaction` and handle this inside the function?

[the remark this answers is not in the document]
Comment (MHHukiewitz) [reply src/pages/data/components/BrowseDataTable.tsx:41]: Would it make sense to allow both Transaction objects and strings as inputs to `signTransaction` and handle this inside the function?

[the remark this answers is not in the document]
Comment (MHHukiewitz) [reply src/pages/data/components/BrowseDataTable.tsx:41]: Would it make sense to allow both Transaction objects and strings as inputs to `signTransaction` and handle this inside the function?

[the remark this answers is not in the document]
Comment (MHHukiewitz) [reply src/pages/data/components/BrowseDataTable.tsx:41]: Would it make sense to allow both Transaction objects and strings as inputs to `signTransaction` and handle this inside the function?

[the remark this answers is not in the document]
Comment (MHHukiewitz) [reply src/pages/data/components/BrowseDataTable.tsx:41]: Would it make sense to allow both Transaction objects and strings as inputs to `signTransaction` and handle this inside the function?

[the remark this answers is not in the document]
Comment (ricardocr987) [reply src/pages/data/components/BrowseDataTable.tsx:41]: signTransaction is a function from solana wallet adapter and accepts only a transaction objects (Versioned or Legacy Transaction)

**Review by manomage — APPROVED** [review]

**Pull request merged**